File: js/programmers/level1/예산.js
function solution(d, budget) {
  d.sort((a, b) => a - b);
  let cnt = 0;
  for (let i = 0; i < d.length; i++) {
    if (budget - d[i] >= 0) {
      budget -= d[i];
      cnt += 1;
    }
  }
  return cnt;
}

console.log(solution([1, 3, 2, 5, 4], 9));
console.log(solution([2, 2, 3, 3], 10));

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        Arrays.sort(d);
        int result = 0;
        for (int i=0; i<d.length; i++) {
            if (budget - d[i] >= 0) {
                budget -= d[i];
                result += 1;
                continue;
            }
            break;
        }
        return result;
    }
}